### DISCUSSION TITLE
SQLAlchemy and DeclarativeBase subclass

### DISCUSSION THREAD

**Discussion by bjdm** [pallets-eco/flask-sqlalchemy#1135] [Q&A]
I have been playing around with SQLAlchemy 2.0 and was wondering what the idiomatic way to correctly provide metadata for table creation is now? Currently I am passing Base.metadata as a keyword parameter to SQLAlchemy when creating but the metadata of some tables is missing and I've been unable to troubleshoot why.

Moving forward, will automatically sourcing the metadata from DeclarativeBase be something flask-sqlalchemy will do itself?

```python
from flask_sqlalchemy import SQLAlchemy
from sqlalchemy.orm import DeclarativeBase

class Base(DeclarativeBase):
    pass

db = SQLAlchemy(metadata=Base.metadata)
```

**Comment by davidism** [comment]
In most cases, you shouldn't define your own base. If you do, you don't need to pass the metadata, you need to pass the base. Make sure to derive the base from Flask-SQLAlchemy's `Model` and `DefaultMeta` so that you get the features that Flask-SQLAlchemy provides.

```python
db = SQLAlkchemy(model_class=Base)
```
